src/main/java/nextstep/subway/line/domain/SectionLineUp.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    private boolean isEndUpSection(Section section) {
+        return section.isSameDownSection(findUpStation());
+    }
+
+    private boolean isEndDownSection(Section section) {
+        return section.isSameUpSection(findDownStation());
+    }
+
+    private void isAlreadyExistSection(StationLineUp stationLineUp, Section section) {
+        if (stationLineUp.stationExisted(section.getUpStation()) &&
+                stationLineUp.stationExisted(section.getDownStation())) {
+            throw new IllegalArgumentException("이미 등록된 구간 입니다.");
+        }
+    }
+
+    private void isNotExistSection(StationLineUp stationLineUp, Section section) {
+        if (!stationLineUp.isEmpty() && stationLineUp.unKnownStation(section.getUpStation())
+                && stationLineUp.unKnownStation(section.getDownStation())) {
+            throw new IllegalArgumentException("등록할 수 없는 구간 입니다.");
+        }
+    }
+
+    public List<Station> getStations() {
+        if (this.sections.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Station> stations = new ArrayList<>();
+        Station downStation = findUpStation();
+        stations.add(downStation);
+
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Optional<Section> nextLineStation = sections.stream()
+                    .filter(it -> it.getUpStation() == finalDownStation)
+                    .findFirst();
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            downStation = nextLineStation.get().getDownStation();
+            stations.add(downStation);
+        }
+
+        return stations;
+    }
+
+    private Station findUpStation() {
+        Station downStation = sections.get(FIRST_INDEX).getUpStation();
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Optional<Section> nextLineStation = sections.stream()
+                    .filter(it -> it.getDownStation().equals(finalDownStation))
+                    .findFirst();
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            downStation = nextLineStation.get().getUpStation();
+        }
+
+        return downStation;
+    }
+
+    private Station findDownStation() {
+        Station upStation = sections.get(FIRST_INDEX).getDownStation();
+        while (upStation != null) {
+            Station finalUpStation = upStation;
+            Optional<Section> nextLineStation = sections.stream()
+                    .filter(it -> it.getUpStation().equals(finalUpStation))
+                    .findFirst();
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            upStation = nextLineStation.get().getDownStation();
+        }
+
+        return upStation;
+    }
+
+    public void remove(Station station) {
+        if (this.sections.size() <= 1) {
```

Comment:
```suggestion
        if (this.sections.size() <= MIN_SIZE) {
```

매직넘버 상수화를 해보아도 좋을것 같네요!

Reply:
제가 놓친 부분인데 잘 찾아주셨네요 !! 👍
피드백 내용 반영하여 상수로 선언 및 사용하도록 개선하였습니다
감사합니다 !